src/components/Modals/DeleteExpenseModal.js

```diff
@@ -0,0 +1,67 @@
+import { useContext, useState, forwardRef, useImperativeHandle } from 'react';
```

Comment:
i am not sure how more you are motivated to tinker with the refactoring of modals. i could imagine that you want to be done with it at some point. so please feel free to just leave everything as is. it is a good solution (as much as react allows it to be). however, if you want to use it further as a play/learn opportunity, here more ideas:

goal: make modals as simple as alerts.
idea: what if all modals can take an _optional_ expense. this way, we could eliminate the need for the need for the delete modal.


alternative: have you tinkered with the idea of inheritance/classes?

```js
class Modal // basic stuff
class ConfirmMonthModal extends Modal // overrides just the title and contents
class ExpenseDeleteModal extends Modal // overrides just the title and contents
// ...
```

Reply:
Yeah, I'm not feeling super motivated to continue tinkering with this, and some of the solutions I've looked into are mind-boggling to me, even if they are meant to be elegant and useful. I'm not too keen are adding code I don't understand. I might spend some time on the side trying to wrap my head around these alternatives, and then bring the issue up again in the future if/when I feel comfortable exploring our own implementation.

For now, one change (slight improvement?) we could make is to move the modal components, refs, functions out of `UserDashboard` and into the respective components that use/open/call them so they are separate and localized. So we'd have `ChangeMonthModal` inside `AddExpenses`, and `EditExpenseModal` and `DeleteExpenseModal` inside `Expense`.

Reply:
sounds like a good plan.